sqltap/sqltap.py

```diff
@@ -38,19 +46,30 @@ class QueryStats(object):
     :attr user_context: The value returned by the user_context_fn set
         with :func:`sqltap.start`.
     """
-    def __init__(self, text, stack, duration, user_context):
+    def __init__(self, text, stack, start_time, end_time,
+                 user_context, params_dict, results):
         self.text = text
-        self.params = getattr(text, 'params', {})
+        self.params = params_dict
+        self.params_id = None
         self.stack = stack
-        self.duration = duration
+        self.start_time = start_time
+        self.end_time = end_time
+        self.duration = end_time - start_time
         self.user_context = user_context
+        self.rowcount = results.rowcount
+        self.params_hash = self._calculate_params_hash(self.params)
 
-        if self.params is None:
-            self.params = {}
+    def _calculate_params_hash(self, params):
+        h = 0
+        for k in sorted(params.keys()):
+            h ^= 10009 * hash(params[k])
+        return (h ^ (h >> 32)) & ((1 << 32) - 1)  # convert to 32-bit unsigned
```

Comment:
I might be reading this wrong, but I *believe* that by only including the value and not scoping this per-query, when you display the values, you'll end up double counting parameter sets. For example, with two different queries that have filters `WHERE user_id = :user_id` and `WHERE account_id = :account_id`, if `:user_id` == 12 and `:account_id` == 12, won't that muck up the calculation?